src/LinuxMain.ts

```diff
@@ -0,0 +1,48 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the VSCode Swift open source project
+//
+// Copyright (c) 2022 the VSCode Swift project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of VSCode Swift project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import * as vscode from "vscode";
+import { FolderContext } from "./FolderContext";
+import { pathExists } from "./utilities/utilities";
+
+/**
+ * Cache the existence of `Tests/LinuxMain.swift`.
+ */
+export class LinuxMain implements vscode.Disposable {
+    private fileWatcher: vscode.FileSystemWatcher;
+    public exists?: boolean;
+
+    constructor(private folderContext: FolderContext) {
+        pathExists(this.folderContext.folder.fsPath, "Tests", "LinuxMain.swift").then(
```

Comment:
This change would make all `hasLinuxMain` dependent `async` IMO.  I’m wondering if it’s worth it.
Another solution is to allow `hasLinuxMain` return `undefined` and let the caller decide what to do. This will add to complexity at call site.

Reply:
If you call `LinuxMain.create()` in `WorkspaceContext.create()` and pass the fully built `LinuxMain` into the `WorkspaceContext` constructor, anyone accessing `linuxMain` via `WorkspaceContext` won't have to deal with an undefined `hasLinuxMain`